airflow/utils/json.py

```diff
@@ -59,7 +59,11 @@ def default(self, o: Any) -> Any:
             data = serialize(o)
             if isinstance(data, dict) and DATA in data:
                 return data[DATA]
-
+        if isinstance(o, bytes):
+            try:
+                return o.decode()
+            except UnicodeDecodeError:
+                return repr(o)
```

Comment:
An alternative would be to use `o.decode("unicode_escape")`, which renders undecodable characters as `\uXXXX`. This is probably better than leaking the `b"..."` format.